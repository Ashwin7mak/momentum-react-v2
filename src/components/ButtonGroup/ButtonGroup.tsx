import React, { FC } from 'react';

import { DEFAULTS, STYLE } from './ButtonGroup.constants';
import { Props } from './ButtonGroup.types';
import './ButtonGroup.style.scss';

const ButtonGroup: FC<Props> = (props: Props) => {
  let children = Array.isArray(props.children) ? [...props.children] : [props.children];

  children = children.reduce((newChildren, child, index) => {
    if (index !== 0) {
      newChildren.push(<div key={`separator-${index}`} className={STYLE.separator} />);
    }

    newChildren.push(child);

    return newChildren;
  }, []);

  const classes = props.className ? `${STYLE.wrapper} ${props.className}` : STYLE.wrapper;

  return (
    <div
      className={classes}
      data-round={props.round || DEFAULTS.ROUND}
      data-separation={props.separation || DEFAULTS.SEPARATION}
      data-spaced={props.spaced || DEFAULTS.SPACED}
    >
      {children}
    </div>
  );
};

export default ButtonGroup;
