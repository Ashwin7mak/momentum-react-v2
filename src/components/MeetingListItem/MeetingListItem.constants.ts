const CLASS_PREFIX = 'md-meeting-list-item';

const DEFAULTS = {};

const STYLE = {
  wrapper: `${CLASS_PREFIX}-wrapper`,
};

export { CLASS_PREFIX, DEFAULTS, STYLE };
