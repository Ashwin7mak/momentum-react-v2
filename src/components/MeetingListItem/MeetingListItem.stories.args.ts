import { commonStyles } from '../../storybook/helper.stories.argtypes';

export default {
  ...commonStyles,
  children: {
    description: 'Provides the child nodes for this element.', // Description of this prop.
    control: { type: 'none' }, // Control
    table: {
      type: {
        summary: 'ReactNode', // Explicit type of this prop.
      },
      defaultValue: {
        summary: 'undefined', // Default value when rendering this component.
      },
    },
  },
  isDisabled: {
    defaultValue: false,
    description: 'Determines if this item is disabled',
    control: { type: 'boolean' },
    table: {
      type: {
        summary: 'boolean',
      },
      defaultValue: {
        summary: false,
      },
    },
  },
  buttonGroup: {
    defaultValue: undefined,
    description: 'Provides the nodes for the end of the item',
    control: { type: 'none' },
    table: {
      type: {
        summary: 'ReactNode',
      },
      defaultValue: {
        summary: undefined,
      },
    },
  },
  color: {
    defaultValue: undefined,
    description: 'Provides the color status of this item',
    control: { type: 'select' },
    table: {
      type: {
        summary: "'join' | 'inactive' | 'activeNoJoin' | 'empty'",
      },
      defaultValue: {
        summary: undefined,
      },
    },
  },
  startImage: {
    defaultValue: undefined,
    description: 'Provides the nodes for the start of the item',
    control: { type: 'none' },
    table: {
      type: {
        summary: 'ReactNode',
      },
      defaultValue: {
        summary: undefined,
      },
    },
  },
};
