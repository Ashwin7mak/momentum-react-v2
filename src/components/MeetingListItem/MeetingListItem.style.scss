.md-meeting-list-item-wrapper {
    margin: 1rem 0;
    padding-right: 0.5rem;
    
    &[data-color="join"],
    &[data-color="inactive"],
    &[data-color="activeNoJoin"] {
        background-color:  rgba(255, 255, 255, 0.11); // IEFIX
        background-color: var(--theme-background-primary-active);

        &[data-disabled="true"], &.disable {
            background-color:  rgba(255, 255, 255, 0.11); // IEFIX
            background-color: var(--theme-background-primary-active);
        }
    }

    &[data-color="empty"] {
        background-color:  rgba(255, 255, 255, 0.11); // IEFIX
        background-color: var(--theme-background-primary-active);

        &:hover, &.hover, &[data-disabled="true"], &.disable {
            background-color:  rgba(255, 255, 255, 0.11);  // IEFIX
            background-color: var(--theme-background-primary-active);
        }
    }

    .md-meeting-list-item-start-section {
        display: flex;
        align-items: center;
        .md-meeting-list-item-border {
            height: 3.125rem;
            width: 0.3rem;
            border-radius: 0.5rem 0 0 0.5rem;
            margin-right: 0.5rem;

            &[data-color="join"] {
                background-color: rgba(29, 128, 95, 1); // IEFIX
                background-color: var(--button-join-fill-background);
            }

            &[data-color="activeNoJoin"] {
                background-color: rgba(0, 0, 0 , 0.2); // IEFIX
                background-color: var(--button-secondary-border-disabled);
            }
            
            &[data-color="inactive"] {

                background: webkit-repeating-linear-gradient(
                    -45deg,
                    transparent,
                    transparent 7%,
                    rgba(255, 255, 255, 0.3) 7%,
                    rgba(255, 255, 255, 0.3) 14%
                );

                background: -o-repeating-linear-gradient(
                    -45deg,
                    transparent,
                    transparent 7%,
                    rgba(255, 255, 255, 0.3) 7%,
                    rgba(255, 255, 255, 0.3) 14%
                );

                background: repeating-linear-gradient(
                    -45deg,
                    transparent,
                    transparent 7%,
                    rgba(255, 255, 255, 0.3) 7%,
                    rgba(255, 255, 255, 0.3) 14%
                );
            }
        }
    }
}
