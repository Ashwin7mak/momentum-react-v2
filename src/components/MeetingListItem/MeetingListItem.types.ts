import { CSSProperties, ReactNode } from 'react';

export interface Props {
  /**
   * Buttons or icons for end of item
   */
  buttonGroup?: ReactNode;

  /**
   * Child components of this ButtonPill.
   */
  children?: ReactNode;

  /**
   * Custom class for overriding this component's CSS.
   */
  className?: string;

  /**
   * Color status
   */
  color?: 'join' | 'inactive' | 'activeNoJoin' | 'empty';

  /**
   * Custom id for overriding this component's CSS.
   */
  id?: string;

  /**
   * Determines if this item is disabled
   * @default false
   */
  isDisabled?: boolean;

  /**
   * Icon, Avatar, or other content for beginning of item
   */
  image?: ReactNode;

  /**
   * Custom style for overriding this component's CSS.
   */
  style?: CSSProperties;
}
