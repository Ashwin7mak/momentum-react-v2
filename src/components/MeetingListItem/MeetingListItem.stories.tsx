import React from 'react';
import {
  getComponentStates,
  MultiTemplate,
  MultiTemplateWithPseudoStates,
  Template,
} from '../../storybook/helper.stories.templates';
import { DocumentationPage } from '../../storybook/helper.stories.docs';
import StyleDocs from '../../storybook/docs.stories.style.mdx';

import MeetingListItem, { MeetingListItemProps } from './';
import argTypes from './MeetingListItem.stories.args';
import Documentation from './MeetingListItem.stories.docs.mdx';
import Icon from '../Icon';
import Avatar from '../Avatar';
import ButtonHyperlink from '../ButtonHyperlink';
import ButtonPill from '../ButtonPill';
import ButtonGroup from '../ButtonGroup';
import ButtonCircle from '../ButtonCircle';

export default {
  title: 'Momentum UI/MeetingListItem',
  component: MeetingListItem,
  parameters: {
    expanded: true,
    docs: {
      page: DocumentationPage(Documentation, StyleDocs),
    },
  },
  args: {
    // Args provided to all stories by default.
    children: 'A', // Example of a default arg for all stories.
  },
};

/**
 * Primary story. This renders a single component with all external props.
 */
const Example = Template((args) => (
  <>
    {getComponentStates(
      MeetingListItem,
      {
        ...args,
        children: (
          <>
            <div key="child1">Date</div>
            <div key="child2">Normal</div>
          </>
        ),
      },
      {}
    )}
  </>
)).bind({});

Example.argTypes = { ...argTypes };

Example.args = {
  buttonGroup: (
    <ButtonGroup spaced>
      <ButtonHyperlink>Link</ButtonHyperlink>
      <div style={{ display: 'flex', alignItems: 'center' }}>
        17 <Icon autoScale name="participant-list" />
      </div>
      <ButtonPill color="join">Join</ButtonPill>
    </ButtonGroup>
  ),
  children: (
    <>
      <div key="child1">Date</div>
      <div key="child2">Normal</div>
    </>
  ),
  color: 'join',
  startImage: <Avatar initials="TU" size={32} />,
};

/**
 * Common variants story. This renders multiple variants of a single component.
 */
const Common = MultiTemplateWithPseudoStates(MeetingListItem).bind({});

Common.argTypes = { ...argTypes };
delete Common.argTypes.children;

Common.parameters = {
  variants: [
    {
      label: 'Recording with buttons',
      buttonGroup: (
        <ButtonGroup spaced>
          <Icon autoScale name="info-circle" />
          <Icon autoScale name="share-c-native-iph" />
          <Icon autoScale name="chat" />
        </ButtonGroup>
      ),
      children: (
        <>
          <div>Today</div>
          <div>11:00 - 12:00</div>
        </>
      ),
      startImage: (
        <ButtonCircle>
          <Icon autoScale name="play" />
        </ButtonCircle>
      ),
    },
    {
      label: 'Recording with buttons and link',
      buttonGroup: (
        <ButtonGroup spaced>
          <ButtonHyperlink>hyperlink</ButtonHyperlink>
          <Icon autoScale name="info-circle" />
          <Icon autoScale name="chat" />
        </ButtonGroup>
      ),
      children: (
        <>
          <div>Today</div>
          <div>11:00 - 12:00</div>
        </>
      ),
      startImage: (
        <ButtonCircle>
          <Icon autoScale name="play" />
        </ButtonCircle>
      ),
    },
    {
      label: 'Active meeting with participant count',
      buttonGroup: (
        <ButtonGroup spaced>
          <ButtonHyperlink>Link</ButtonHyperlink>
          <div style={{ display: 'flex', alignItems: 'center' }}>
            17 <Icon autoScale name="participant-list" />
          </div>
          <ButtonPill color="join">Join</ButtonPill>
        </ButtonGroup>
      ),
      children: (
        <>
          <div key="child1">Date</div>
          <div key="child2">Normal</div>
        </>
      ),
      color: 'join',
      startImage: <Avatar initials="TU" size={32} />,
    },
    {
      label: 'Inactive meeting',
      buttonGroup: (
        <ButtonGroup spaced>
          <Icon autoScale name="recurring" />
          <Icon autoScale name="calendar-empty" />
        </ButtonGroup>
      ),
      children: (
        <>
          <div key="child1">Date</div>
          <div key="child2">Normal</div>
        </>
      ),
      color: 'inactive',
      startImage: <Avatar initials="TU" size={32} />,
    },
    {
      label: 'Active meeting without join button',
      buttonGroup: (
        <ButtonGroup spaced>
          <Icon autoScale name="recurring" />
          <Icon autoScale name="calendar-empty" />
        </ButtonGroup>
      ),
      children: (
        <>
          <div key="child1">Date</div>
          <div key="child2">Normal</div>
        </>
      ),
      color: 'activeNoJoin',
      startImage: <Avatar initials="TU" size={32} />,
    },
    {
      label: 'No meetings scheduled',
      children: 'No meetings scheduled',
      color: 'empty',
    },
    {
      label: 'Scheduled meeting',
      children: (
        <div style={{ display: 'flex' }}>
          <div style={{ display: 'flex', flexDirection: 'column' }}>
            <div>Date</div>
            <div>Time - Time</div>
          </div>
          <div style={{ display: 'flex', alignItems: 'center', marginLeft: '10px' }}>
            <div style={{ marginRight: '5px' }}>Normal</div> <Icon autoScale name="recurring" />
          </div>
        </div>
      ),
      buttonGroup: <Avatar initials="TU" size={32} />,
    },
    {
      label: 'Scheduled meeting',
      children: (
        <div style={{ display: 'flex' }}>
          <div style={{ display: 'flex', flexDirection: 'column' }}>
            <div>Date</div>
            <div>Time - Time</div>
          </div>
          <div style={{ display: 'flex', alignItems: 'center', marginLeft: '10px' }}>
            <div style={{ marginRight: '5px' }}>Normal</div> <Icon autoScale name="recurring" />
          </div>
        </div>
      ),
      buttonGroup: <Avatar initials="TU" size={32} />,
      isDisabled: true,
    },
  ],
};

const Colors = MultiTemplate<MeetingListItemProps>(MeetingListItem).bind({});

Colors.argTypes = { ...argTypes };
delete Colors.argTypes.children;

Colors.parameters = {
  variants: [
    {
      buttonGroup: (
        <ButtonGroup spaced>
          <ButtonHyperlink>Link</ButtonHyperlink>
        </ButtonGroup>
      ),
      children: <div>Join</div>,
      color: 'join',
      startImage: <Avatar initials="TU" size={32} />,
    },
    {
      buttonGroup: (
        <ButtonGroup spaced>
          <ButtonHyperlink>Link</ButtonHyperlink>
        </ButtonGroup>
      ),
      children: <div>Inactive</div>,
      color: 'inactive',
      startImage: <Avatar initials="TU" size={32} />,
    },
    {
      buttonGroup: (
        <ButtonGroup spaced>
          <ButtonHyperlink>Link</ButtonHyperlink>
        </ButtonGroup>
      ),
      children: <div>Active No Join</div>,
      color: 'activeNoJoin',
      startImage: <Avatar initials="TU" size={32} />,
    },
  ],
};

export { Example, Common, Colors };
