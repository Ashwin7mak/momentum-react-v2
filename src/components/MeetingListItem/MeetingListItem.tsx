import React, { Children, cloneElement, FC, ReactElement } from 'react';
import classnames from 'classnames';

import { STYLE } from './MeetingListItem.constants';
import { Props } from './MeetingListItem.types';
import './MeetingListItem.style.scss';
import ListItemBase from '../ListItemBase';
import ListItemBaseSection from '../ListItemBaseSection';
import { verifyTypes } from '../../helpers/verifyTypes';
import ButtonPill from '../ButtonPill';
import ButtonCircle from '../ButtonCircle';
import Avatar from '../Avatar';
import Icon from '../Icon';

const MeetingListItem: FC<Props> = (props: Props) => {
  const { children, className, color, isDisabled, buttonGroup, image, large, ...rest } = props;

  const getSizeProps = (element: ReactElement) => {
    const sizeProps = {};
    if (verifyTypes(element, ButtonPill)) {
      sizeProps['size'] = 28;
    } else if (verifyTypes(element, ButtonCircle)) {
      sizeProps['size'] = 32;
    } else if (verifyTypes(element, Icon)) {
      sizeProps['autoScale'] = true;
    } else if (verifyTypes(element, Avatar)) {
      sizeProps['size'] = 32;
    }
    return sizeProps;
  };

  const getSizedElement = (element) => {
    if (element && React.isValidElement(element)) {
      const elementChildren = element.props['children'];
      let sizedChildren;
      if (elementChildren) {
        if (Array.isArray(elementChildren)) {
          sizedChildren = [];
          Children.map(elementChildren, (child) => {
            sizedChildren.push(getSizedElement(child));
          });
        } else {
          sizedChildren = getSizedElement(elementChildren);
        }
      }
      return cloneElement(element, getSizeProps(element), sizedChildren);
    }
    return element;
  };

  let buttonSection = buttonGroup;

  if (buttonGroup && React.isValidElement(buttonGroup)) {
    buttonSection = getSizedElement(buttonGroup);
  }

  const middleSectionColorClass = color ? ` md-meeting-list-item-middle-section-${color}` : '';

  return (
    <ListItemBase
      className={`${classnames(className, STYLE.wrapper)}`}
      data-color={color}
      isDisabled={isDisabled}
      size={large ? 70 : 50}
      {...rest}
    >
      <ListItemBaseSection className="md-meeting-list-item-start-section" position="start">
        <div className="md-meeting-list-item-border" data-color={color} />
        {React.isValidElement(image) ? getSizedElement(image) : image}
      </ListItemBaseSection>

      <ListItemBaseSection
        className={`md-meeting-list-item-middle-section${middleSectionColorClass}`}
        position="middle"
      >
        {getSizedElement(children)}
      </ListItemBaseSection>

      {buttonSection && (
        <ListItemBaseSection className="md-meeting-list-item-end-section" position="end">
          {buttonSection}
        </ListItemBaseSection>
      )}
    </ListItemBase>
  );
};

export default MeetingListItem;
