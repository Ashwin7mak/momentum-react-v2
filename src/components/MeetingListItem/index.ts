import { default as MeetingListItem } from './MeetingListItem';
import * as CONSTANTS from './MeetingListItem.constants';
import { Props, MeetingMarker } from './MeetingListItem.types';

export { CONSTANTS as MEETING_LIST_ITEM_CONSTANTS, MeetingMarker };
export type MeetingListItemProps = Props;

export default MeetingListItem;
