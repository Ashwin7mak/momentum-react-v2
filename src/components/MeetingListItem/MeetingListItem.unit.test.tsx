import React from 'react';
import { mount } from 'enzyme';

import MeetingListItem, { MEETING_LIST_ITEM_CONSTANTS as CONSTANTS } from './';
import ButtonGroup from '../ButtonGroup';
import Icon from '../Icon';

describe('<MeetingListItem />', () => {
  describe('snapshot', () => {
    it('should match snapshot', () => {
      expect.assertions(1);

      const container = mount(<MeetingListItem />);

      expect(container).toMatchSnapshot();
    });

    it('should match snapshot with className', () => {
      expect.assertions(1);

      const className = 'example-class';

      const container = mount(<MeetingListItem className={className} />);

      expect(container).toMatchSnapshot();
    });

    it('should match snapshot with id', () => {
      expect.assertions(1);

      const id = 'example-id';

      const container = mount(<MeetingListItem id={id} />);

      expect(container).toMatchSnapshot();
    });

    it('should match snapshot with style', () => {
      expect.assertions(1);

      const style = { color: 'pink' };

      const container = mount(<MeetingListItem style={style} />);

      expect(container).toMatchSnapshot();
    });

    it('should match snapshot with isDisabled', () => {
      expect.assertions(1);

      const isDisabled = true;

      const container = mount(<MeetingListItem isDisabled={isDisabled} />);

      expect(container).toMatchSnapshot();
    });

    it('should match snapshot with buttonGroup', () => {
      expect.assertions(1);

      const buttonGroup = <ButtonGroup />;

      const container = mount(<MeetingListItem buttonGroup={buttonGroup} />);

      expect(container).toMatchSnapshot();
    });

    it('should match snapshot with color', () => {
      expect.assertions(1);

      const color = 'join';

      const container = mount(<MeetingListItem color={color} />);

      expect(container).toMatchSnapshot();
    });

    it('should match snapshot with startImage', () => {
      expect.assertions(1);

      const startImage = <Icon name="placeholder" />;

      const container = mount(<MeetingListItem startImage={startImage} />);

      expect(container).toMatchSnapshot();
    });
  });

  describe('attributes', () => {
    it('should have its wrapper class', () => {
      expect.assertions(1);

      const element = mount(<MeetingListItem />)
        .find(MeetingListItem)
        .getDOMNode();

      expect(element.classList.contains(CONSTANTS.STYLE.wrapper)).toBe(true);
    });

    it('should have provided class when className is provided', () => {
      expect.assertions(1);

      const className = 'example-class';

      const element = mount(<MeetingListItem className={className} />)
        .find(MeetingListItem)
        .getDOMNode();

      expect(element.classList.contains(className)).toBe(true);
    });

    it('should have provided id when id is provided', () => {
      expect.assertions(1);

      const id = 'example-id';

      const element = mount(<MeetingListItem id={id} />)
        .find(MeetingListItem)
        .getDOMNode();

      expect(element.id).toBe(id);
    });

    it('should have provided style when style is provided', () => {
      expect.assertions(1);

      const style = { color: 'pink' };
      const styleString = 'color: pink;';

      const element = mount(<MeetingListItem style={style} />)
        .find(MeetingListItem)
        .getDOMNode();

      expect(element.getAttribute('style')).toBe(styleString);
    });

    it('should have provided data-disabled when isDisabled is provided', () => {
      expect.assertions(1);

      const isDisabled = true;

      const container = mount(<MeetingListItem isDisabled={isDisabled}>Test</MeetingListItem>);

      const element = container.find(MeetingListItem).getDOMNode();

      expect(element.getAttribute('data-disabled')).toBe('true');
    });

    it('should have provided data-size', () => {
      expect.assertions(1);

      const container = mount(<MeetingListItem>Test</MeetingListItem>);

      const element = container.find(MeetingListItem).getDOMNode();

      expect(element.getAttribute('data-size')).toBe('50');
    });

    it('should have provided data-shape', () => {
      expect.assertions(1);

      const container = mount(<MeetingListItem>Test</MeetingListItem>);

      const element = container.find(MeetingListItem).getDOMNode();

      expect(element.getAttribute('data-shape')).toBe('rectangleNoPadding');
    });

    it('should have provided role', () => {
      expect.assertions(1);

      const container = mount(<MeetingListItem>Test</MeetingListItem>);

      const element = container.find(MeetingListItem).getDOMNode();

      expect(element.getAttribute('role')).toBe('listitem');
    });
  });
});
