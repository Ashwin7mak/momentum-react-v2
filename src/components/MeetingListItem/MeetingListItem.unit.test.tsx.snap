// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MeetingListItem /> snapshot should match snapshot 1`] = `
<MeetingListItem>
  <ListItemBase
    className="md-meeting-list-item-wrapper"
    shape="rectangleNoPadding"
    size={50}
  >
    <FocusRing>
      <FocusRing
        focusClass="md-focus-ring-wrapper children"
        focusRingClass="md-focus-ring-wrapper children"
      >
        <li
          className="md-meeting-list-item-wrapper md-list-item-base-wrapper"
          data-disabled={false}
          data-shape="rectangleNoPadding"
          data-size={50}
          onBlur={[Function]}
          onFocus={[Function]}
          role="listitem"
        >
          <ListItemBaseSection
            className="md-meeting-list-item-start-section"
            position="start"
          >
            <div
              className="md-meeting-list-item-start-section"
              data-position="start"
            >
              <div
                className="md-meeting-list-item-border"
              />
            </div>
          </ListItemBaseSection>
          <ListItemBaseSection
            position="middle"
          >
            <div
              data-position="middle"
            />
          </ListItemBaseSection>
        </li>
      </FocusRing>
    </FocusRing>
  </ListItemBase>
</MeetingListItem>
`;

exports[`<MeetingListItem /> snapshot should match snapshot with buttonGroup 1`] = `
<MeetingListItem
  buttonGroup={<ButtonGroup />}
>
  <ListItemBase
    className="md-meeting-list-item-wrapper"
    shape="rectangleNoPadding"
    size={50}
  >
    <FocusRing>
      <FocusRing
        focusClass="md-focus-ring-wrapper children"
        focusRingClass="md-focus-ring-wrapper children"
      >
        <li
          className="md-meeting-list-item-wrapper md-list-item-base-wrapper"
          data-disabled={false}
          data-shape="rectangleNoPadding"
          data-size={50}
          onBlur={[Function]}
          onFocus={[Function]}
          role="listitem"
        >
          <ListItemBaseSection
            className="md-meeting-list-item-start-section"
            position="start"
          >
            <div
              className="md-meeting-list-item-start-section"
              data-position="start"
            >
              <div
                className="md-meeting-list-item-border"
              />
            </div>
          </ListItemBaseSection>
          <ListItemBaseSection
            position="middle"
          >
            <div
              data-position="middle"
            />
          </ListItemBaseSection>
          <ListItemBaseSection
            position="end"
          >
            <div
              data-position="end"
            >
              <ButtonGroup>
                <div
                  className="md-button-group-wrapper"
                  data-round={false}
                  data-spaced={false}
                />
              </ButtonGroup>
            </div>
          </ListItemBaseSection>
        </li>
      </FocusRing>
    </FocusRing>
  </ListItemBase>
</MeetingListItem>
`;

exports[`<MeetingListItem /> snapshot should match snapshot with className 1`] = `
<MeetingListItem
  className="example-class"
>
  <ListItemBase
    className="example-class md-meeting-list-item-wrapper"
    shape="rectangleNoPadding"
    size={50}
  >
    <FocusRing>
      <FocusRing
        focusClass="md-focus-ring-wrapper children"
        focusRingClass="md-focus-ring-wrapper children"
      >
        <li
          className="example-class md-meeting-list-item-wrapper md-list-item-base-wrapper"
          data-disabled={false}
          data-shape="rectangleNoPadding"
          data-size={50}
          onBlur={[Function]}
          onFocus={[Function]}
          role="listitem"
        >
          <ListItemBaseSection
            className="md-meeting-list-item-start-section"
            position="start"
          >
            <div
              className="md-meeting-list-item-start-section"
              data-position="start"
            >
              <div
                className="md-meeting-list-item-border"
              />
            </div>
          </ListItemBaseSection>
          <ListItemBaseSection
            position="middle"
          >
            <div
              data-position="middle"
            />
          </ListItemBaseSection>
        </li>
      </FocusRing>
    </FocusRing>
  </ListItemBase>
</MeetingListItem>
`;

exports[`<MeetingListItem /> snapshot should match snapshot with color 1`] = `
<MeetingListItem
  color="join"
>
  <ListItemBase
    className="md-meeting-list-item-wrapper"
    data-color="join"
    shape="rectangleNoPadding"
    size={50}
  >
    <FocusRing>
      <FocusRing
        focusClass="md-focus-ring-wrapper children"
        focusRingClass="md-focus-ring-wrapper children"
      >
        <li
          className="md-meeting-list-item-wrapper md-list-item-base-wrapper"
          data-color="join"
          data-disabled={false}
          data-shape="rectangleNoPadding"
          data-size={50}
          onBlur={[Function]}
          onFocus={[Function]}
          role="listitem"
        >
          <ListItemBaseSection
            className="md-meeting-list-item-start-section"
            position="start"
          >
            <div
              className="md-meeting-list-item-start-section"
              data-position="start"
            >
              <div
                className="md-meeting-list-item-border"
                data-color="join"
              />
            </div>
          </ListItemBaseSection>
          <ListItemBaseSection
            position="middle"
          >
            <div
              data-position="middle"
            />
          </ListItemBaseSection>
        </li>
      </FocusRing>
    </FocusRing>
  </ListItemBase>
</MeetingListItem>
`;

exports[`<MeetingListItem /> snapshot should match snapshot with id 1`] = `
<MeetingListItem
  id="example-id"
>
  <ListItemBase
    className="md-meeting-list-item-wrapper"
    id="example-id"
    shape="rectangleNoPadding"
    size={50}
  >
    <FocusRing>
      <FocusRing
        focusClass="md-focus-ring-wrapper children"
        focusRingClass="md-focus-ring-wrapper children"
      >
        <li
          className="md-meeting-list-item-wrapper md-list-item-base-wrapper"
          data-disabled={false}
          data-shape="rectangleNoPadding"
          data-size={50}
          id="example-id"
          onBlur={[Function]}
          onFocus={[Function]}
          role="listitem"
        >
          <ListItemBaseSection
            className="md-meeting-list-item-start-section"
            position="start"
          >
            <div
              className="md-meeting-list-item-start-section"
              data-position="start"
            >
              <div
                className="md-meeting-list-item-border"
              />
            </div>
          </ListItemBaseSection>
          <ListItemBaseSection
            position="middle"
          >
            <div
              data-position="middle"
            />
          </ListItemBaseSection>
        </li>
      </FocusRing>
    </FocusRing>
  </ListItemBase>
</MeetingListItem>
`;

exports[`<MeetingListItem /> snapshot should match snapshot with isDisabled 1`] = `
<MeetingListItem
  isDisabled={true}
>
  <ListItemBase
    className="md-meeting-list-item-wrapper"
    isDisabled={true}
    shape="rectangleNoPadding"
    size={50}
  >
    <FocusRing>
      <FocusRing
        focusClass="md-focus-ring-wrapper children"
        focusRingClass="md-focus-ring-wrapper children"
      >
        <li
          className="md-meeting-list-item-wrapper md-list-item-base-wrapper"
          data-disabled={true}
          data-shape="rectangleNoPadding"
          data-size={50}
          onBlur={[Function]}
          onFocus={[Function]}
          role="listitem"
        >
          <ListItemBaseSection
            className="md-meeting-list-item-start-section"
            position="start"
          >
            <div
              className="md-meeting-list-item-start-section"
              data-position="start"
            >
              <div
                className="md-meeting-list-item-border"
              />
            </div>
          </ListItemBaseSection>
          <ListItemBaseSection
            position="middle"
          >
            <div
              data-position="middle"
            />
          </ListItemBaseSection>
        </li>
      </FocusRing>
    </FocusRing>
  </ListItemBase>
</MeetingListItem>
`;

exports[`<MeetingListItem /> snapshot should match snapshot with startImage 1`] = `
<MeetingListItem
  startImage={
    <Icon
      name="placeholder"
    />
  }
>
  <ListItemBase
    className="md-meeting-list-item-wrapper"
    shape="rectangleNoPadding"
    size={50}
  >
    <FocusRing>
      <FocusRing
        focusClass="md-focus-ring-wrapper children"
        focusRingClass="md-focus-ring-wrapper children"
      >
        <li
          className="md-meeting-list-item-wrapper md-list-item-base-wrapper"
          data-disabled={false}
          data-shape="rectangleNoPadding"
          data-size={50}
          onBlur={[Function]}
          onFocus={[Function]}
          role="listitem"
        >
          <ListItemBaseSection
            className="md-meeting-list-item-start-section"
            position="start"
          >
            <div
              className="md-meeting-list-item-start-section"
              data-position="start"
            >
              <div
                className="md-meeting-list-item-border"
              />
              <Icon
                name="placeholder"
              />
            </div>
          </ListItemBaseSection>
          <ListItemBaseSection
            position="middle"
          >
            <div
              data-position="middle"
            />
          </ListItemBaseSection>
        </li>
      </FocusRing>
    </FocusRing>
  </ListItemBase>
</MeetingListItem>
`;

exports[`<MeetingListItem /> snapshot should match snapshot with style 1`] = `
<MeetingListItem
  style={
    Object {
      "color": "pink",
    }
  }
>
  <ListItemBase
    className="md-meeting-list-item-wrapper"
    shape="rectangleNoPadding"
    size={50}
    style={
      Object {
        "color": "pink",
      }
    }
  >
    <FocusRing>
      <FocusRing
        focusClass="md-focus-ring-wrapper children"
        focusRingClass="md-focus-ring-wrapper children"
      >
        <li
          className="md-meeting-list-item-wrapper md-list-item-base-wrapper"
          data-disabled={false}
          data-shape="rectangleNoPadding"
          data-size={50}
          onBlur={[Function]}
          onFocus={[Function]}
          role="listitem"
          style={
            Object {
              "color": "pink",
            }
          }
        >
          <ListItemBaseSection
            className="md-meeting-list-item-start-section"
            position="start"
          >
            <div
              className="md-meeting-list-item-start-section"
              data-position="start"
            >
              <div
                className="md-meeting-list-item-border"
              />
            </div>
          </ListItemBaseSection>
          <ListItemBaseSection
            position="middle"
          >
            <div
              data-position="middle"
            />
          </ListItemBaseSection>
        </li>
      </FocusRing>
    </FocusRing>
  </ListItemBase>
</MeetingListItem>
`;
